### PR TITLE
Add Kafka Example

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?

This example shows how to collect kafka metrics + kafka logs using two options:
- JMX Receiver + kafka metrics receiver
- JMX Metrics Gatherer + kafka metrics receiver

Using this example will populate the OOTB Kafka Dashboard.

### CODE REVIEW

File: apps/kafka-metrics/consumer/app/src/main/java/consumer/App.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                            BatchLogRecordProcessor.builder(
+                                    OtlpGrpcLogRecordExporter.builder()
+                                        .setEndpoint("http://otelcol:4317")
+                                        .build())
+                                .build())
+                        .build())
+                .build();
+    
+        // Add hook to close SDK, which flushes logs
+        Runtime.getRuntime().addShutdownHook(new Thread(sdk::close));
```
[the remark this answers is not in the document]
Comment (mackjmr) [reply apps/kafka-metrics/consumer/app/src/main/java/consumer/App.java]: Yes, updated in last commit 👍

File: apps/kafka-metrics/consumer/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    implementation 'org.apache.kafka:kafka-clients:2.3.0'
+
+    // logger
+    implementation(platform("org.apache.logging.log4j:log4j-bom:2.22.1"))
+    implementation("org.apache.logging.log4j:log4j-api")
+    implementation("org.apache.logging.log4j:log4j-core")
+    implementation("org.apache.logging.log4j:log4j-slf4j2-impl:2.22.1")
+
+    // OpenTelemetry core
+    implementation 'io.opentelemetry:opentelemetry-sdk:1.34.1'
```
[the remark this answers is not in the document]
Comment (mackjmr) [reply apps/kafka-metrics/consumer/app/build.gradle]: Removed

File: apps/kafka-metrics/producer/app/src/main/java/producer/App.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                                .put("service.name", "log4j-example")
+                                .build())
+                        .addLogRecordProcessor(
+                            BatchLogRecordProcessor.builder(
+                                    OtlpGrpcLogRecordExporter.builder()
+                                        .setEndpoint("http://otelcol:4317")
+                                        .build())
+                                .build())
+                        .build())
+                .build();
```
[the remark this answers is not in the document]
Comment (mackjmr) [reply apps/kafka-metrics/producer/app/src/main/java/producer/App.java]: Updated to use auto-instrumentation

File: apps/kafka-metrics/Dockerfile.collector [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+# JMX Metrics Gatherer Jar
+ARG JMX_GATHERER_JAR_VERSION=1.27.0
+ADD https://github.com/open-telemetry/opentelemetry-java-contrib/releases/download/v${JMX_GATHERER_JAR_VERSION}/opentelemetry-jmx-metrics.jar /opt/opentelemetry-jmx-metrics.jar
+# nonroot user id (https://groups.google.com/g/distroless-users/c/-DpzCr7xRDY/m/eQqJmJroCgAJ)
+ARG USER_UID=65532
+RUN chown ${USER_UID} /opt/opentelemetry-jmx-metrics.jar
+
+
+FROM gcr.io/distroless/java11-debian11:nonroot
```
Comment (dineshg13) [inline apps/kafka-metrics/Dockerfile.collector:17]: can we use java 17 ?

File: apps/kafka-metrics/consumer/app/src/main/java/consumer/App.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
```
Comment (dineshg13) [inline apps/kafka-metrics/consumer/app/src/main/java/consumer/App.java:18]: Remove and use appender in log4j directly

**Pull request merged**